View a list of articles

elvisiraguha opened the pull request:
# Description

This PR contains changes that set up firebase collection and fetch articles from firebase collection.

## Type of change

Please check the options that are relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How should this be manually tested or reproduced?

- Clone this repo
- Checkout to `ft-fetch-articles-firebase`
- Run the command `sensible-browser UI/pages/blogs.html`

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules

# Screenshots (if applicable)
![shot-2020-08-03_20-53-13](https://user-images.githubusercontent.com/34744026/89216871-7cf91f00-d5cb-11ea-9c41-1dc267e4b51e.jpg)

elvisiraguha committed:
feature(firebase): view a list of articles

- set up firebase store for this application
- add a collection of articles
- add a collection of comments
- fetch articles and comments from firebase
- post a new comment on firebase collection

bdushimi commented on UI/pages/article.html:
```diff
@@ -97,22 +97,40 @@ <h4>Comments:</h4>
+     https://firebase.google.com/docs/web/setup#available-libraries -->
+    <script src="https://www.gstatic.com/firebasejs/7.17.1/firebase-analytics.js"></script>
+
+    <script>
```
@elvisiraguha it would be better to have this firebase config and initialization scripts to a reusable js file so that it can be imported whenever another file/piece of codes need access to the firestore.

bdushimi commented on UI/scripts/blogs.js:
```diff
@@ -148,5 +137,13 @@ const paginationButton = (page, items) => {
+    setUpPagination(snapshot.docs, paginationSection, rows);
+  })
+  .catch((err) => {
+    console.log(err);
```
@elvisiraguha can you also display this error to the user? not in the console.

bdushimi commented on UI/scripts/article.js:
```diff
@@ -213,23 +196,44 @@ const handleSubmit = (e) => {
+  .then((doc) => {
+    const data = doc.data();
+    displayArticle(data);
+  });
```
@elvisiraguha it would be safer to handle errors just in case the currentArticleId is not correct or does not exist in the db,

elvisiraguha committed:
feature(firebase): display error messages correctly

- display errors to the user instead of console
- have firebase configuration in one file that is accessible to other pages

elvisiraguha committed:
separate display notification for easy re-use
